torchvision/prototype/transforms/_augment.py

```diff
@@ -380,7 +324,73 @@ def forward(self, *inputs: Any) -> Any:
             output_images.append(output_image)
             output_targets.append(output_target)
 
-        # Insert updated images and targets into input flat_sample
-        self._insert_outputs(flat_inputs, output_images, output_targets)
+        return unflatten_and_insert_data(flat_batch_with_spec, output_images, output_targets)
+
+
+class MixupDetection(_DetectionBatchTransform):
+    _transformed_types = (is_simple_tensor, datapoints.Image, PIL.Image)
+
+    def __init__(
+        self,
+        *,
+        alpha: float = 1.5,
+    ) -> None:
+        super().__init__()
+        self._dist = torch.distributions.Beta(torch.tensor([alpha]), torch.tensor([alpha]))
+
+    def _check_inputs(self, flat_inputs: List[Any]) -> None:
+        if has_any(flat_inputs, datapoints.Mask, datapoints.Video):
+            raise TypeError(f"{type(self).__name__}() is only supported for images and bounding boxes.")
+
+    def _get_params(self, flat_inputs: List[Any]) -> Dict[str, Any]:
+        return dict(ratio=float(self._dist.sample()))
+
+    def forward(self, *inputs: Any) -> Any:
+        flat_batch_with_spec, batch = self._flatten_and_extract_data(
+            inputs,
+            image=(datapoints.Image, PIL.Image.Image, is_simple_tensor),
+            boxes=(datapoints.BoundingBox,),
+            labels=(datapoints.Label, datapoints.OneHotLabel),
+        )
+        self._check_inputs(flat_batch_with_spec[0])
+
+        batch = self._to_image_tensor(batch)
+
+        batch_output = [
+            self._mixup(sample, sample_rolled, self._get_params([])["ratio"])
+            for sample, sample_rolled in zip(batch, batch[-1:] + batch[:-1])
+        ]
+
+        return self._unflatten_and_insert_data(flat_batch_with_spec, batch_output)
+
+    def _mixup(self, sample_1: Dict[str, Any], sample_2: Dict[str, Any], ratio: float) -> Dict[str, Any]:
+        if ratio >= 1.0:
+            return sample_1
+        elif ratio == 0.0:
+            return sample_2
+
+        h_1, w_1 = sample_1["image"].shape[-2:]
+        h_2, w_2 = sample_2["image"].shape[-2:]
+        h_mixup = max(h_1, h_2)
+        w_mixup = max(w_1, w_2)
+
+        # TODO: add the option to fill this with something else than 0
+        dtype = sample_1["image"].dtype if sample_1["image"].is_floating_point() else torch.float32
+        mix_image = F.pad_image_tensor(
+            sample_1["image"].to(dtype), padding=[0, 0, w_mixup - w_1, h_mixup - h_1], fill=None
+        ).mul_(ratio)
+        mix_image[..., :h_2, :w_2] = sample_2["image"] * (1.0 - ratio)
+        mix_image = mix_image.to(sample_1["image"])
```

Comment:
You commented this out and indeed there was a bug in my implementation. I fixed that so we can reinstate it?